test.py

```diff
@@ -1,124 +1,527 @@
+    def setUp(self):
+        self.error_parser = ServoErrorLogParser()
+        self.multi_log = open('resources/multi-line-comment.log').read()
+        self.expected_multi_errors = error_sample
+        self.single_log = open('resources/single-line-comment.log').read()
+        self.expected_single_errors = [
+            {
+                'body': 'missing space before {', 
+                'position': 49, 
+                'path': './components/plugins/lints/sorter.rs'
+            }
+        ]
+
+
+    def test_parse_errors(self):
+        self.assertEqual(self.expected_multi_errors, list(self.error_parser.parse_log(self.multi_log)))
+        self.assertEqual(self.expected_single_errors, list(self.error_parser.parse_log(self.single_log)))
+
+
+class TestPayloadHandler(unittest.TestCase):
+    def setUp(self):
+        self.handler = PayloadHandler()
+
+
+    def test_handle_payload(self):
+        with self.assertRaises(NotImplementedError):
+            self.handler.handle_payload("payload")
+
+
+class TestTravisPayloadHandler(unittest.TestCase):    
+    def setUp(self):
+        class TravisDouble():
+            def get_build(self, build_id):
+                return 1
+
+
+            def get_log(self, build_data):
+                return open('resources/multi-line-comment.log').read()
+
+
+            def get_pull_request_number(self, build_data):
+                return 1
+
+
+        class ErrorParserDouble():
+            path_key = ServoErrorLogParser.path_key
+            body_key = ServoErrorLogParser.body_key
+            position_key = ServoErrorLogParser.position_key
+
+            def parse_log(self, log):
+                return error_sample
+
+
+        travis_dbl = TravisDouble()
+        error_parser_dbl = ErrorParserDouble()
+        self.github = GithubApiProvider("jdm", "a453b3923e893f0383cd2893f...", "servo", "servo")
+        self.github.post_review_comment = Mock()
+        self.github.get_review_comments = Mock(return_value=json.loads(open('resources/review_comments.json').read()))
+
+        self.payload_handler = TravisPayloadHandler(self.github, travis_dbl, error_parser_dbl)
+        
+
+    def test_handle_payload(self):
+        payload = json.loads(open('resources/test_travis_payload.json').read())
+        self.payload_handler.handle_payload(payload)
+        err_msg = TravisPayloadHandler.msg_template
+
+        self.github.post_review_comment.assert_called_with(1, "9b6313fd5ab92de5a3fd9f13f8421a929b2a8ef6", err_msg.format(error_sample[3]['path'], error_sample[3]['position'], error_sample[3]['body']), error_sample[3]['path'], error_sample[3]['position'])
+
+class TestGithubPayloadHandler(unittest.TestCase):
+    def setUp(self):
+        self.github_user = "jdm"
+        self.github = GithubApiProvider(self.github_user, "a453b3923e893f0383cd2893f...", "servo", "servo")
+        self.github.remove_label = Mock()
+        self.github.add_label = Mock()
+
+
+    def test_handle_payload(self):
```

Comment:
This feels like it's testing implementation details that are not particularly important. To me, the benefit of the integration tests that existed before were that they verified the output of the bot, given a particular input. These particular tests feel oddly coupled to the actual implementation strategy to me.